nomad/structs/config/ui.go

```diff
@@ -24,6 +24,30 @@ type UIConfig struct {
 	Label *LabelUIConfig `hcl:"label"`
 }
 
+// only covers the elements of
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP we need or care about
+type ContentSecurityPolicy struct {
+	ConnectSrc     []string `hcl:"connect_src"`
+	DefaultSrc     []string `hcl:"default_src"`
+	FormAction     []string `hcl:"form_action"`
+	FrameAncestors []string `hcl:"frame_ancestors"`
+	ImgSrc         []string `hcl:"img_src"`
+	ScriptSrc      []string `hcl:"script_src"`
+	StyleSrc       []string `hcl:"style_src"`
+}
+
+func DefaultCSPConfig() *ContentSecurityPolicy {
+	return &ContentSecurityPolicy{
+		ConnectSrc:     []string{"*"},
+		DefaultSrc:     []string{"none"},
+		FormAction:     []string{"none"},
+		FrameAncestors: []string{"none"},
+		ImgSrc:         []string{"self", "data:"},
+		ScriptSrc:      []string{"self"},
+		StyleSrc:       []string{"self", "unsafe-inline"},
```

Comment:
I mentioned in my previous comment that we'd need to figure out the quoting for these when we render the header value. The simply `fmt` interpolation you did is much easier, so that's fine, but now we have to make sure the default values are quoted properly. Looking at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources#sources I think a bunch of these need to be single-quoted inside the string, because they're special values.

```suggestion
		DefaultSrc:     []string{"'none'"},
		FormAction:     []string{"'none'"},
		FrameAncestors: []string{"'none'"},
		ImgSrc:         []string{"'self'", "data:"},
		ScriptSrc:      []string{"'self'"},
		StyleSrc:       []string{"'self'", "'unsafe-inline'"},
```